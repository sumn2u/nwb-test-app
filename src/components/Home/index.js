import React, { Component } from 'react';
import styled from 'styled-components'

const Home = ({ txt }) => {

    const HelloWorld = styled.div`
    color: ${txt == 'of course it works' ? 'green' : 'red'};
    font-size: 18px;
    border: solid 2px black;
    padding: 20px;
    margin-top: 20px;
    `

    return <HelloWorld>{txt}</HelloWorld>;
};

export default Home;
