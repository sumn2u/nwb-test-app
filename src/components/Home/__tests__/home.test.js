import React from 'react';
import Home  from '../index';
import {
    shallow,
    mount,
    render
} from 'enzyme';

describe('Home Component', () => {
    it('should render correctly the home component', () => {
        const component = shallow( < Home / > );
        expect(component).toMatchSnapshot();
    });
});


describe('test', () => {
    it('should work', () => {
        const wrapper = shallow( < Home txt = "is working" / > );
    });
});