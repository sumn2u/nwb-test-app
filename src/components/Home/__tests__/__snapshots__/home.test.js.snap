// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render correctly the home component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    color: ",
          "red",
          ";
    font-size: 18px;
    border: solid 2px black;
    padding 20px;
    margin-top: 20px;
    ",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    color: ",
            "red",
            ";
    font-size: 18px;
    border: solid 2px black;
    padding 20px;
    margin-top: 20px;
    ",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
